packages/devtools_app/lib/src/screens/network/network_controller.dart

```diff
@@ -168,52 +165,34 @@ class NetworkController extends DisposableController
   bool get isPolling => _pollingTimer != null;
 
   void _initHelper() async {
-    try {
-      if (offlineDataController.showingOfflineData.value == true) {
-        await maybeLoadOfflineData(
-          NetworkScreen.id,
-          createData: (json) => OfflineNetworkData.fromJson(json),
-          shouldLoad: (data) => !data.isEmpty,
-          loadData: (data) => loadOfflineData(data),
-        );
-      }
-      cancelListeners();
-      if (!offlineDataController.showingOfflineData.value) {
-        unawaited(startRecording());
-        debugPrint('started recording');
-        addAutoDisposeListener(
-          serviceConnection.serviceManager.isolateManager.mainIsolate,
-          () {
-            if (serviceConnection
-                    .serviceManager.isolateManager.mainIsolate.value !=
-                null) {
-              unawaited(startRecording());
-            }
-          },
-        );
-      }
-    } catch (e) {
-      _log.shout('Could not load offline data: $e');
+    if (offlineDataController.showingOfflineData.value) {
+      await maybeLoadOfflineData(
+        NetworkScreen.id,
+        createData: (json) => OfflineNetworkData.fromJson(json),
+        shouldLoad: (data) => !data.isEmpty,
+        loadData: (data) => loadOfflineData(data),
+      );
+    } else {
+      await startRecording();
     }
   }
 
   Future<void> loadOfflineData(OfflineNetworkData offlineData) async {
-    // Set filtered data to the requests available in offline data.
-    filteredData
+    _currentNetworkRequests
       ..clear()
-      ..addAll(offlineData.requests);
+      ..updateOrAddAll(
+        requests: offlineData.currentRequests!,
+        sockets: offlineData.socketStats,
+        timelineMicrosOffset: DateTime.now().microsecondsSinceEpoch -
+            (networkService.timeStamp ?? 0),
```

Comment:
You can just use `DateTime.now().microsecondsSinceEpoch` for this. This is arbitrary for offline data since we won't be updating the sockets like we do when we have a live connection.

Reply:
done